tap_google_sheets/streams.py

```diff
@@ -1,74 +1,519 @@
+                stream_name=stream_name,
+                record=record,
+                time_extracted=time_extracted)
+    except OSError as err:
+        LOGGER.info('OS Error writing record for: {}'.format(stream_name))
+        raise err
+
+def get_bookmark(state, stream, default):
+    """
+    Get bookmark for the stream
+    """
+    if (state is None) or ('bookmarks' not in state):
+        return default
+    return (
+        state
+        .get('bookmarks', {})
+        .get(stream, default)
+    )
+
+def write_bookmark(state, stream, value):
+    """
+    Write bookmark for the stream
+    """
+    if 'bookmarks' not in state:
+        state['bookmarks'] = {}
+    state['bookmarks'][stream] = value
+    LOGGER.info('Write state for stream: {}, value: {}'.format(stream, value))
+    singer.write_state(state)
+
+def get_abs_path(path):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
+
+# List selected fields from stream catalog
+def get_selected_fields(catalog, stream_name):
+    """
+    Get the selected fields for a stream from the catalog
+    """
+    stream = catalog.get_stream(stream_name)
+    mdata = metadata.to_map(stream.metadata)
+    mdata_list = singer.metadata.to_list(mdata)
+    selected_fields = []
+    for entry in mdata_list:
+        field = None
+        try:
+            field = entry['breadcrumb'][1]
+            if entry.get('metadata', {}).get('selected', False):
+                selected_fields.append(field)
+        except IndexError:
+            pass
+    return selected_fields
+
+class GoogleSheets:
+    stream_name = None
+    api = None
+    path = None
+    key_properties = None
+    replication_method = None
+    replication_keys = None
+    params = None
+    state = None
+
+    def __init__(self, client, spreadsheet_id, start_date=None):
+        self.client = client
+        self.config_start_date = start_date
+        self.spreadsheet_id = spreadsheet_id
+
+    # return schema for streams
+    def get_schemas(self):
+        schemas = {}
+        field_metadata = {}
+
+        schema_path = get_abs_path('schemas/{}.json'.format(self.stream_name))
+        with open(schema_path) as file:
+            schema = json.load(file)
+        schemas[self.stream_name] = schema
+        mdata = metadata.new()
+
+        # Documentation:
+        # https://github.com/singer-io/getting-started/blob/master/docs/DISCOVERY_MODE.md#singer-python-helper-functions
+        # Reference:
+        # https://github.com/singer-io/singer-python/blob/master/singer/metadata.py#L25-L44
+        mdata = metadata.get_standard_metadata(
+            schema=schema,
+            key_properties=self.key_properties,
+            valid_replication_keys=self.replication_keys,
+            replication_method=self.replication_method
+        )
+        field_metadata[self.stream_name] = mdata
+
+        return schemas, field_metadata
+
+    # Transform/validate batch of records w/ schema and sent to target
```

Comment:
What w/ means?

Reply:
Please keep consistency in function comments, add below the def fun_name in all functions.

Reply:
Added comments below the function declaration.